tendrl/node_agent/manager/manager.py

```diff
@@ -239,6 +246,31 @@ def on_pull(self, raw_data):
                     self.etcd_client.write(("nodes/%s/Disks/free/%s") % (
                         raw_data["node_id"], disk), "")
 
+    def load_and_execute_discovery_plugins(self, node_id):
```

Comment:
I feel we can have generic module defined at tendrl.node_agent.discovery.manager.py which defines a function say `execute_discoveries()`. This class would define individual functions for loading different types of plugins and these functions would be called from inside the __init__() function of manager. Inside execute_discoveries we would execute each type of plugins one by one.

This class could look something like 
```
class DiscoveryManager(object):
    def __init__(self):
        try:
            self.load_platform_discovery_plugins()
            self.load_sds_discovery_plugins()
        except (SyntaxError, ValueError, ImportError) as ex:
            LOG.error('Failed to load the platform plugins. Error %s' %
                       ex, exc_info=True)
            raise ex
    
    def load_platform_discovery_plugins(self):
        ..........

    def load_sds_discovery_plugins(self):
        .............

    def execute_discoveries(self):
        for plugin in PlatformDiscoveryPlugin.plugins:
            # execute the function say discover_platform() and add details to etcd

        for plugin in SDSDiscoveryPlugin.plugins:
            # execute the function say discover_sds() and add details to etcd
```

From node agent manager all we need to do is 

`DiscoveryManager().execute_discoveries()` in `main()`